src/Storages/StorageMergeTree.cpp

```diff
@@ -1094,6 +1095,7 @@ std::shared_ptr<MergeMutateSelectedEntry> StorageMergeTree::selectPartsToMutate(
 
             current_ast_elements += commands_size;
             commands->insert(commands->end(), it->second.commands.begin(), it->second.commands.end());
+            commands->settings_changes = it->second.commands.settings_changes;
```

Comment:
The same issue as in `ReplicatedMergeTreeQueue::getMutationCommands` (btw, it takes settings from the last mutation entry, but for `ReplicatedMergeTree` it takes setting from the first entry). Consider adding a test that runs multiple mutations with different settings asynchronously.